File: src/35. 搜索插入位置/res.java
class Solution {
    public int searchInsert(int[] nums, int target) {
      int l = 0, r = nums.length - 1, max = nums.length;
      while (l < r){
        int mid = l + r >> 1;
        if (nums[mid] >= target) r = mid;
        else l = mid + 1;
      }

      if (target > nums[r]) return max;
      else return l;
    }
}

File: src/binary-search/res.cpp
class Solution
{
public:
  int search(vector<int> &nums, int target)
  {
    int l = 0;
    int r = nums.size() - 1;
    while (l < r)
    {
      int mid = (l + r) / 2;
      if (nums[mid] >= target)
      {
        r = mid;
      }
      else
      {
        l = mid + 1;
      }
    }
    if (nums[l] == target)
    {
      return l;
    }
    return -1;
  }
};